src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java

```diff
@@ -0,0 +1,308 @@
+            Timestamp interval_end_time;
+
+            // Check if experiment_name is provided
+            if (experiment_name == null || experiment_name.isEmpty()) {
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.EXPERIMENT_NAME_MANDATORY);
+                return;
+            }
+
+            // Check if interval_end_time is provided
+            if (intervalEndTimeStr == null || intervalEndTimeStr.isEmpty()) {
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.INTERVAL_END_TIME_MANDATORY);
+                return;
+            }
+            if (!Utils.DateUtils.isAValidDate(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT, intervalEndTimeStr)) {
+                interval_end_time = null;
+                sendErrorResponse(
+                        response,
+                        new Exception(AnalyzerErrorConstants.APIErrors.ListRecommendationsAPI.INVALID_TIMESTAMP_EXCPTN),
+                        HttpServletResponse.SC_BAD_REQUEST,
+                        String.format(AnalyzerErrorConstants.APIErrors.ListRecommendationsAPI.INVALID_TIMESTAMP_MSG, intervalEndTimeStr)
+                );
+                return;
+            } else {
+                interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT, intervalEndTimeStr);
+            }
+            LOGGER.debug("experiment_name : {} and interval_end_time : {} ", experiment_name,  intervalEndTimeStr);
+
+            List<ExperimentResultData> experimentResultDataList = new ArrayList<>();
+            ExperimentResultData experimentResultData = null;
+            Map<String, KruizeObject> mainKruizeExperimentMAP = new ConcurrentHashMap<>();
+            KruizeObject kruizeObject = null;
+            try {
+                new ExperimentDBService().loadExperimentFromDBByName(mainKruizeExperimentMAP, experiment_name);
+                if (null != mainKruizeExperimentMAP.get(experiment_name)) {
+                    kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
+                }
+                // Get DataSource and connect
+                // Get MetricsProfile name and list of promQL to fetch
+                // Find LongTerm duration  keep Start , End and Step(measurement Duration)
+                // prepare <List>ExperimentResultData
+                // and call new ExperimentInitiator().generateAndAddRecommendations(kruizeObject, experimentResultDataList, interval_start_time, interval_end_time);    // TODO: experimentResultDataList not required
+
+                List<String> promQls = new ArrayList<>();
+                promQls.add("sum(rate(container_cpu_usage_seconds_total{container!~\"POD|\",namespace=\"%s\",container=\"%s\"}[%sm])) by (namespace,container)");
```

Comment:
This Promqls will be constant and will be derived from Metrics profile , 16 more promql will be added